packages/block-library/src/image/image.js

```diff
@@ -960,12 +962,20 @@ export default function Image( {
 				onResizeStart={ onResizeStart }
 				onResizeStop={ ( event, direction, elt ) => {
 					onResizeStop();
+
+					let resizedWidth = `${ elt.offsetWidth }px`;
+					// Set width to 'auto' if the resized width is close to the max-content width.
+					if ( Math.abs( elt.offsetWidth - maxContentWidth ) < 10 ) {
+						resizedWidth = 'auto';
+						elt.style.width = 'auto';
```

Comment:
The default state of an image when it's inserted is that the controls display `auto`, but there's no explicit style output. Instead of injecting `width: 'auto'` and `height: auto`, could we instead treat this as a kind of "reset" and an early return, with something like:

```js
setAttributes( {
  width: undefined,
  height: undefined,
  aspectRatio: undefined,
} );
return;
```

If we go with what we currently have, I notice that the markup winds up including the `auto` rules:

```
<!-- wp:image {"id":4711,"width":"auto","height":"auto","sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large is-resized"><img src="http://a-toule-site.local/wp-content/uploads/2024/01/IMG_3906-768x1024.jpg" alt="" class="wp-image-4711" style="width:auto;height:auto"/></figure>
```

Whereas an image that is freshly inserted doesn't have any of those rules applied:

```
<!-- wp:image {"id":4711,"sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large"><img src="http://a-toule-site.local/wp-content/uploads/2024/01/IMG_3906-768x1024.jpg" alt="" class="wp-image-4711"/></figure>
```

Reply:
Good idea unsetting to `undefined`!

I don't know if it's possible to set an image to 100% of the container's width, so I added a check to fallback to explicit pixel values if the image is smaller than the container.

Reply:
Thanks Kai, this is feeling really close to me!

> I don't know if it's possible to set an image to 100% of the container's width, so I added a check to fallback to explicit pixel values if the image is smaller than the container.

That sounds good to me for now — we could always look at it in a follow-up, potentially alongside an explicit way of locking to `full width` alignment as I imagine the two concepts could be quite related. E.g. when would we want to lock to "auto" vs setting the alignment 🤔

For now, though, I think unsetting and only doing so if the image is smaller than the container is a good solution 👍

The one issue I ran into in retesting is that I think I was wrong about explicitly setting `aspectRatio` to `undefined`. It looks like we might need to either not update the `aspectRatio` property at all in this call, or reuse the logic from the `setAttributes` call below:

```
						aspectRatio:
							ratio === naturalRatio
								? undefined
								: String( ratio ),
```

Otherwise, when adjusting the size within the threshold of the container area, we lose an aspect ratio if one has been set:

https://github.com/user-attachments/assets/6574397e-c559-4bce-a5e6-373403eb89ef

What do you think? From my perspective, once that bug's resolved I think this PR will be in good shape to switch over to ready for review and get wider testing / feedback from designers, etc.

Reply:
I think that makes sense! Maybe we could just not override the `aspectRatio` in this case? I pushed an update in [c1635bd](https://github.com/WordPress/gutenberg/pull/63341/commits/c1635bd75a2821e339691f9470a3b407aa1f7039).

Reply:
Nice, that's testing well for me now! (Also while re-testing, I think avoiding explicitly setting `auto` for width and height helps avoid potential conflicts with aspect ratios, too). Let's see what some designers think 🙂